src/components/App/App.jsx

```diff
@@ -27,18 +36,114 @@ function App() {
   const [articlesObserv, setObservArticles] = useState(3);
   const [totalArticles, setTotalArticles] = useState(0);
   const [isSerchingStarted, setSerchingStart] = useState(false);
+  const [userArticles, setUserArticles] = useState(null);
+  const [keyword, setKeyword] = useState("");
+
+  
+  const [formValidators, setFormValidators] = useState([]);
+
+  const [currentUser, setCurrentUser] = useState(null);
+  const navigate = useNavigate();
+
+  // Form Validation
+  useEffect(() => {
+    if (isModalRegistraionOpen ||isModalLoginOpen) {
+      const modalsForms = document.querySelectorAll(".modal__form");
```

Comment:
**Could be improved**

`document.querySelector` and `querySelectorAll` is not allowed to use in `React` because it’s a **very slow** operation for the browser. React was specially created not to search anything in the DOM. So, you can directly operate with the markup inside the components. Or you can use hook `useRef` for the reference.


But it's better to implement the validation hook I suggested previously https://github.com/NeysonAdams/newx-explorer--frontend/pull/2#discussion_r1714284566